Add `customers` query

loan-laux opened the pull request:
Signed-off-by: Loan Laux <loan@outgrow.io>

Resolves #12 
Impact: **major**
Type: **feature**

## Issue
There was no way to get all the customer accounts with a GraphQL query.

## Solution
Create a `customers` query which returns all the customers as paginated data.

## Breaking changes
None.


## Testing
1. Create customer accounts through a storefront.
2. Query `customers` with an admin's `Authorization` token.
3. See all of the customer accounts being returned.

loan-laux committed:
feat: add customers query

Signed-off-by: Loan Laux <loan@outgrow.io>

aldeed commented on src/queries/customers.js:
```diff
@@ -0,0 +1,18 @@
+  await context.validatePermissions("reaction:legacy:accounts", "read");
+
+  return Accounts.find({
+    groups: []
```
Might be safer to do `groups: { $in: [null, []] }` to handle `null` or missing `groups` field?

aldeed reviewed (changes requested):
See one inline suggestion, and maybe it would be good to also add support for this in the `accounts` query? For client UIs that are already using that query it might lead to simpler client code.

It could be a new bool param on `accounts` GQL query like

```graphql
notInAnyGroups: Boolean
```

If used along with groupIds, it would return accounts in any of those groups OR in no group.

If it's too much work at the moment, it can be a future PR.

loan-laux committed:
refactor: guard for groups === null

Signed-off-by: Loan Laux <loan@outgrow.io>

loan-laux committed:
feat: add notInAnyGroups arg to accounts query

Signed-off-by: Loan Laux <loan@outgrow.io>

loan-laux commented:
@aldeed Good idea! I've added `notInAnyGroups` to the `accounts` query, and I've added a guard against `null` groups in the customer query. Let me know if you'd like other changes made.

aldeed commented:
Since the notInAnyGroups logic is somewhat complex, it would be good to add some integration tests for it in the `reaction` repo after this is pulled in.

rc-publisher commented:
:tada: This PR is included in version 1.4.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@reactioncommerce/api-plugin-accounts/v/1.4.0)
- [GitHub release](https://github.com/reactioncommerce/api-plugin-accounts/releases/tag/v1.4.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
